src/functions/utils.php

```diff
@@ -26,40 +26,77 @@ function tribe_array_merge_recursive( array &$array1, array &$array2 ) {
 	}
 }
 
+
 if ( ! function_exists( 'tribe_register_plugin' ) ) {
 	/**
 	 * Checks if this plugin has permission to run, if not it notifies the admin
 	 *
-	 * @param string $file_path   Full file path to the base plugin file
-	 * @param string $main_class  The Main/base class for this plugin
-	 * @param string $version     The version
-	 * @param array  $classes_req Any Main class files/tribe plugins required for this to run
+	 * @param string $file_path    Full file path to the base plugin file
+	 * @param string $main_class   The Main/base class for this plugin
+	 * @param string $version      The version
+	 * @param array  $classes_req  Any Main class files/tribe plugins required for this to run
+	 * @param array  $dependencies an array of dependencies to check
 	 *
 	 * @return bool Indicates if plugin should continue initialization
 	 */
-	function tribe_register_plugin( $file_path, $main_class, $version, $classes_req = array() ) {
-		$tribe_dependency = Tribe__Dependency::instance();
-		$should_plugin_run = true;
-
-		// Checks to see if the plugins are active
-		if ( ! empty( $classes_req ) && ! $tribe_dependency->has_requisite_plugins( $classes_req ) ) {
-			$should_plugin_run = false;
+	function tribe_register_plugin( $file_path, $main_class, $version, $classes_req = array(), $dependencies = array() ) {
+		$tribe_dependency  = Tribe__Dependency::instance();
 
-			$tribe_plugins = new Tribe__Plugins();
-			$admin_notice  = new Tribe__Admin__Notice__Plugin_Download( $file_path );
+		//add all plugins to registered_plugins
+		$tribe_dependency->add_registered_plugin( $main_class, $version, $file_path, $dependencies );
 
-			foreach ( $classes_req as $class => $version ) {
+		// Checks to see if the plugins are active for extensions
+		if ( ! empty( $classes_req ) && ! $tribe_dependency->has_requisite_plugins( $classes_req ) ) {
+			$tribe_plugins     = new Tribe__Plugins();
+			$admin_notice      = new Tribe__Admin__Notice__Plugin_Download( $file_path );
+			foreach ( $classes_req as $class => $plugin_version ) {
 				$plugin    = $tribe_plugins->get_plugin_by_class( $class );
-				$is_active = $tribe_dependency->is_plugin_version( $class, $version );
+				$is_active = $tribe_dependency->is_plugin_version( $class, $plugin_version );
 				$admin_notice->add_required_plugin( $plugin['short_name'], $plugin['thickbox_url'], $is_active );
 			}
 		}
 
-		if ( $should_plugin_run ) {
+		// only set The Events Calendar and Event Tickets to Active when registering
+		if ( 'Tribe__Events__Main' === $main_class || 'Tribe__Tickets__Main' === $main_class ) {
 			$tribe_dependency->add_active_plugin( $main_class, $version, $file_path );
 		}
 
-		return $should_plugin_run;
+	}
+}
+
+if ( ! function_exists( 'tribe_check_plugin' ) ) {
+	function tribe_check_plugin( $main_class ) {
+		$tribe_dependency    = Tribe__Dependency::instance();
+		$parent_dependencies = $co_dependencies = $addon_dependencies = false;
```

Comment:
Like the above function, let's put the meat of this logic into `Tribe__Dependency`.